runtime/marshaler.go

```diff
@@ -1,20 +1,27 @@
 package runtime
 
 import (
+	"context"
 	"io"
 )
 
-// Marshaler defines a conversion between byte sequence and gRPC payloads / fields.
+// Marshaler defines a outbound conversion between byte sequence and gRPC payloads / fields.
 type Marshaler interface {
 	// Marshal marshals "v" into byte sequence.
-	Marshal(v interface{}) ([]byte, error)
+	Marshal(ctx context.Context, v interface{}) ([]byte, error)
+	// NewEncoder returns an Encoder which writes bytes sequence into "w".
```

Comment:
```suggestion
	// NewEncoder returns an Encoder which writes byte sequences into "w".
```

Reply:
I think `byte sequence` makes more sense here, because it is a single sequence of bytes per call, WDYT?

Reply:
I think in the general case that this `interface` describes, we want an encoder that _can_ write an arbitrary number of byte sequences (for example, when streaming).

Reply:
@johanbrandhorst I see. Commited these changes in https://github.com/grpc-ecosystem/grpc-gateway/pull/1228/commits/35d0c348983bd14912d292a75dbbac47803569a1